summingbird-scalding-test/src/test/scala/com/twitter/summingbird/scalding/BatchedStoreProperties.scala

```diff
@@ -0,0 +1,124 @@
+    Arbitrary {
+      for {
+        l <- Arbitrary.arbitrary[T]
+        u <- Arbitrary.arbitrary[T]
+      } yield Interval.leftClosedRightOpen(l, u)
+    }
+
+  implicit val arbTimestamp: Arbitrary[Timestamp] = Arbitrary {
+    // a relevant 200 or so year range
+    Gen.choose(1L, 100000L)
+      .map { Timestamp(_) }
+  }
+
+  def sample[T: Arbitrary]: T = Arbitrary.arbitrary[T].sample.get
+
+  implicit def arbitraryPipeFactory: Arbitrary[PipeFactory[(Int, Int)]] = {
+    Arbitrary {
+      Gen.const {
+        StateWithError[(Interval[Timestamp], Mode), List[FailureReason], FlowToPipe[(Int, Int)]] {
+          (timeMode: (Interval[Timestamp], Mode)) =>
+            {
+              val (time: Interval[Timestamp], mode: Mode) = timeMode
+              val a: FlowToPipe[(Int, Int)] = Reader { (fdM: (FlowDef, Mode)) => TypedPipe.from[(Timestamp, (Int, Int))](Seq((Timestamp(10), (2, 3)))) }
+              Right((timeMode, a))
+            }
+        }
+      }
+    }
+  }
+
+  implicit def timeExtractor[T <: (Long, _)] = TestUtil.simpleTimeExtractor[T]
+
+  implicit def arbitraryInputWithTimeStampAndBatcher: Arbitrary[(List[(Long, Int)], Batcher, TestStore[Int, Int])] = Arbitrary {
+    Arbitrary.arbitrary[List[Int]]
+      .map { in => in.zipWithIndex.map { case (item: Int, time: Int) => (time.toLong, item) } }
+      .map { in =>
+        val batcher = TestUtil.randomBatcher(in)
+        val lastTimeStamp = in.size
+        val testStore = TestStore[Int, Int]("test", batcher, sample[Map[Int, Int]], lastTimeStamp)
+        (in, batcher, testStore)
+      }
+  }
+
+  implicit def arbitraryLocalMode: Arbitrary[Mode] = Arbitrary { Gen.const(Local(true)) }
+
+  property("readAfterLastBatch should return interval starting from the last batch written") = {
+    forAll {
+      (
+      a: PipeFactory[(Int, Int)],
+      interval: Interval[Timestamp],
+      inputWithTimeStampAndBatcherAndStore: (List[(Long, Int)], Batcher, TestStore[Int, Int]),
+      mode: Mode
+    ) =>
+        //      println(s"requested interval is ${interval}")
+
+        val (inputWithTimeStamp, batcher, testStore) = inputWithTimeStampAndBatcherAndStore
+
+        val result: PlannerOutput[(BatchID, FlowProducer[TypedPipe[(Int, Int)]], FlowToPipe[(Int, Int)])] = testStore.readAfterLastBatch(a)
+
+        result((interval, mode)) match {
+          case Right(((readInterval: Intersection[InclusiveLower, ExclusiveUpper, Timestamp], _), _)) => {
+            println(s"readInterval is ${readInterval}, intersection is ${interval.intersect(readInterval)}")
+            //readInterval should start from the last written interval
+            val start: InclusiveLower[Timestamp] = InclusiveLower(batcher.earliestTimeOf(testStore.initBatch.next))
+            readInterval.lower == start
+          }
+          case Left(_) => interval == Empty()
+        }
+    }
+  }
+
+  property("should merge Properly") = {
```

Comment:
this is too big of a property. We want smaller ones. Some candidates: the end of the interval is never extended. The output is always ending at the same point as the input, unless the merge cannot be done. What else? Let's think of the properties we expect.